tests/pytests/unit/modules/test_junos.py

```diff
@@ -0,0 +1,2903 @@
+"""
+    :codeauthor: Rajvi Dhimar <rajvidhimar95@gmail.com>
+"""
+import os
+
+import pytest
+import yaml
+
+import salt.modules.junos as junos
+from tests.support.mock import ANY, MagicMock, PropertyMock, call, mock_open, patch
+
+try:
+    from lxml import etree
+except ImportError:
+    import xml.etree.ElementTree as etree
+
+try:
+    import jnpr.junos.op as tables_dir
+    import jxmlease  # pylint: disable=unused-import
+    from jnpr.junos.device import Device
+    from jnpr.junos.exception import ConnectClosedError, LockError, UnlockError
+    from jnpr.junos.utils.config import Config
+    from jnpr.junos.utils.sw import SW
+
+    HAS_JUNOS = True
+except ImportError:
+    HAS_JUNOS = False
+
+pytestmark = [
+    pytest.mark.skip_on_windows(reason="Not supported on Windows"),
+    pytest.mark.skipif(
+        not HAS_JUNOS, reason="The junos-eznc and jxmlease modules are required"
+    ),
+]
+
+
+@pytest.fixture
+def mock_cp(*args, **kwargs):
+    pass
```

Comment:
```suggestion
@pytest.fixture
def mock_cp(*args, **kwargs):
    pass
```
What's this?

A fixture? or a functions that should be used by a mock?

Reply:
The things you pick up when moving tests from unittest to pytest.
But trying to see  the difference, since they look exactly the same to me, what changed ?
Copying the two pieces of code into two files, and get 'Files are identical'

Reply:
It's just a question, no code change.
This should either be a fixture, and, if it is, it's not doing anything, delete it, or a plain function which should be used by a mock, and if it is, its not doing anything, delete it and just use `Mock` or `MagicMock`.

Reply:
I wish you reviewed all the previous tests, in unittest, and then my C & P would be so much easier :laughing: 
Will take a look at the MagicMock since that is only where it is used.

Reply:
> I wish you reviewed all the previous tests, in unittest

You do not wish me good things @dmurphy18 :grin:

Reply:
updated code